### PR TITLE
Add NavBarPopoverButton and NavBarMenuButton

### PR DESCRIPTION
**Pull request by jivey**
https://openstax.atlassian.net/browse/DISCO-14

### CODE REVIEW

File: src/components/NavBarMenuButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+`;
+
+export type NavBarItemProps = React.PropsWithChildren<{
+  label?: string;
+  icon?: string | React.ReactNode;
+  buttonProps?: NavBarButtonProps;
+  popoverProps?: PopoverProps;
+  className?: string;
+}> & (
+  | { label: string; icon?: React.ReactNode; "aria-label"?: string }
```
Comment (Dantemss) [inline src/components/NavBarMenuButtons.tsx:56]: Extra `|` at the beginning of the line here?

Comment (Dantemss) [reply src/components/NavBarMenuButtons.tsx:56]: This entire type can probably be DRYed up. Or at least label and icon can be removed from the stuff inside React.PropsWithChildren. Also, I don't think `icon: string` is allowed in the end based on the types in the OR.

File: src/components/NavBarButton.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { Button, ButtonProps } from "react-aria-components";
+import styled from "styled-components";
+
+export type NavBarButtonProps = {
+  label?: string;
+  icon?: string | React.ReactNode;
+  className?: string;
+  "aria-label"?: string;
+} & ({ label: string } | { "aria-label": string | undefined });
```
Comment (Dantemss) [inline src/components/NavBarButton.tsx:9]: `{ "aria-label": string | undefined }` should be covered by `"aria-label"?: string;` which makes the whole `OR` statement redundant

Comment (Dantemss) [reply src/components/NavBarButton.tsx:9]: I guess the one exception is `"aria-label"?: string;` allows aria-label to not be set, while `"aria-label": string | undefined` requires it to be explicitly set to undefined... that seems like such an odd distinction to make though?

**Review by bethshook — APPROVED** [review]
also curious about the types but otherwise looks good to me

**Comment by jivey** [comment]
@Dantemss @bethshook Thanks for the type comments, I was able to simplify them a great deal if you want to check them out

File: src/components/NavBarButton.tsx (hunk 2 above)
Comment (jivey) [reply src/components/NavBarButton.tsx]: This was to satisfy TS while maintaining full test coverage without using `any` in the spec but in hindsight was over-complicated, spreading the props took it out of the equation.

File: src/components/NavBarMenuButtons.tsx (hunk 1 above)
Comment (jivey) [reply src/components/NavBarMenuButtons.tsx]: I guess prettier likes to put `|` at the beginning of multi line types like this 🤷. Good call about DRYing it up, using the NavBarButtonProps as a base let me delete a lot of type code.

**Review by Dantemss — APPROVED** [review]

**Pull request merged**